src/Compiler/LambdaLift.idr

```diff
@@ -234,6 +297,129 @@ mutual
   liftExp (CErased fc) = pure $ LErased fc
   liftExp (CCrash fc str) = pure $ LCrash fc str
 
+  usedVars : {vars : _} ->
```

Comment:
Can we not pass a `Used vars` around and accumulate the values in it rather than
creating one for each `LLocal` and then merging all of these arrays as we come out
of nodes with multiple arguments?

Reply:
Should be possible (and will probably avoid some unnecessary allocations), I'll get to that one a bit later and update the PR again

Reply:
Done. This change also made the code a bit simpler, because mergeUsed is no longer needed at all.